StortSpelProjekt/Engine/src/Renderer/DX12Tasks/SkyboxRenderTask.cpp

```diff
@@ -0,0 +1,122 @@
+#include "stdafx.h"
+#include "SkyboxRenderTask.h"
+
+#include "../RenderView.h"
+#include "../RootSignature.h"
+#include "../GPUMemory/ConstantBuffer.h"
+#include "../CommandInterface.h"
+#include "../DescriptorHeap.h"
+#include "../SwapChain.h"
+#include "../GPUMemory/Resource.h"
+#include "../PipelineState.h"
+#include "../Renderer/Transform.h"
+#include "../Renderer/Mesh.h"
+#include "../BaseCamera.h"
+#include "../SwapChain.h"
+#include "../GPUMemory/RenderTarget.h"
+#include "../GPUMemory/RenderTargetView.h"
+
+#include "../GPUMemory/ShaderResourceView.h"
+#include "../Texture/TextureCubeMap.h"
+#include "../ECS/Components/SkyboxComponent.h"
+
+SkyboxRenderTask::SkyboxRenderTask(
+	ID3D12Device5* device,
+	RootSignature* rootSignature,
+	LPCWSTR VSName, LPCWSTR PSName,
+	std::vector<D3D12_GRAPHICS_PIPELINE_STATE_DESC*>* gpsds,
+	LPCTSTR psoName)
+	:RenderTask(device, rootSignature, VSName, PSName, gpsds, psoName)
+{
+
+}
+
+SkyboxRenderTask::~SkyboxRenderTask()
+{
+}
+
+void SkyboxRenderTask::SetSkybox(component::SkyboxComponent* skybox)
+{
+	m_pSkybox = skybox;
+}
+
+void SkyboxRenderTask::Execute() 
+{
+	ID3D12CommandAllocator* commandAllocator = m_pCommandInterface->GetCommandAllocator(m_CommandInterfaceIndex);
+	ID3D12GraphicsCommandList5* commandList = m_pCommandInterface->GetCommandList(m_CommandInterfaceIndex);
+	const RenderTargetView* swapChainRenderTarget = m_pSwapChain->GetRTV(m_BackBufferIndex);
+	ID3D12Resource1* swapChainResource = swapChainRenderTarget->GetResource()->GetID3D12Resource1();
+
+	m_pCommandInterface->Reset(m_CommandInterfaceIndex);
+
+	commandList->SetGraphicsRootSignature(m_pRootSig);
+
+	DescriptorHeap* descriptorHeap_CBV_UAV_SRV = m_DescriptorHeaps[DESCRIPTOR_HEAP_TYPE::CBV_UAV_SRV];
+	ID3D12DescriptorHeap* d3d12DescriptorHeap = descriptorHeap_CBV_UAV_SRV->GetID3D12DescriptorHeap();
+	commandList->SetDescriptorHeaps(1, &d3d12DescriptorHeap);
+
+	commandList->SetGraphicsRootDescriptorTable(RS::dtCBV, descriptorHeap_CBV_UAV_SRV->GetGPUHeapAt(0));
+	commandList->SetGraphicsRootDescriptorTable(RS::dtSRV, descriptorHeap_CBV_UAV_SRV->GetGPUHeapAt(0));
+
+	// Change state on front/backbuffer
+	commandList->ResourceBarrier(1, &CD3DX12_RESOURCE_BARRIER::Transition(
+		swapChainResource,
+		D3D12_RESOURCE_STATE_PRESENT,
+		D3D12_RESOURCE_STATE_RENDER_TARGET));
+
+	DescriptorHeap* renderTargetHeap = m_DescriptorHeaps[DESCRIPTOR_HEAP_TYPE::RTV];
+	DescriptorHeap* depthBufferHeap = m_DescriptorHeaps[DESCRIPTOR_HEAP_TYPE::DSV];
+
+	// RenderTargets
+	const unsigned int swapChainIndex = swapChainRenderTarget->GetDescriptorHeapIndex();
+	D3D12_CPU_DESCRIPTOR_HANDLE cdhSwapChain = renderTargetHeap->GetCPUHeapAt(swapChainIndex);
+	D3D12_CPU_DESCRIPTOR_HANDLE cdhs[] = { cdhSwapChain };
+
+	// Depth
+	D3D12_CPU_DESCRIPTOR_HANDLE dsh = depthBufferHeap->GetCPUHeapAt(0);
```

Comment:
Set the mainDSV and use the descriptorHeapIndex from there instead of hardcoding "0"